src/_rtmidi.pyx

```diff
@@ -177,6 +202,11 @@ cdef void _cb_func(double delta_time, vector[unsigned char] *msg_v,
     message = [msg_v.at(i) for i in range(msg_v.size())]
     func((message, delta_time), data)
 
+cdef void _cb_error_func(ErrorType errorType, const string &errorText,
+                         void *cb_info) with gil:
+    """Wrapper for a Python callback function for errors."""
+    func, data, decoder = (<object> cb_info)
+    func(errorType, decoder(errorText, 'auto'), data)
```

Comment:
I suppose the `encoding` parameter of `MidiBase._decode_string` could have a default value if `"auto"`. Seems clenaer to me.